src/Messages.sol

```diff
@@ -174,4 +176,12 @@ library ConnectorMessages {
         amount = uint256(_msg.index(45, 32));
     }
 
+    function formatDomain(Domain domain) public pure returns (bytes9) {
+        return bytes9(byte(uint8(domain)));
+    }
+
+    function formatDomain(Domain domain, uint64 domainId) public pure returns (bytes memory) {
+        return abi.encodePacked(uint8(domain), domainId);
+    }
```

Comment:
We should have both `formatDomain` functions returning `bytes9` to ensure - at the type level - that they both generate a 9-long byte encoded representation of a domain.

```suggestion
    function formatDomain(Domain domain, uint64 domainId) public pure returns (bytes9) {
        return bytes9(abi.encodePacked(uint8(domain), domainId).ref(0).index(0, 9));
    }
```